src/language-server/feature-finder.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Analyzer, Attribute, Document, Element, isPositionInsideRange, Method, ParsedHtmlDocument, Property, SourcePosition, SourceRange} from 'polymer-analyzer';
+import {DatabindingExpression} from 'polymer-analyzer/lib/polymer/expression-scanner';
+
+import {AstLocation, getAstLocationForPosition} from '../ast-from-source-position';
+
+
+/**
+ * Class responsible for describing the object at a given location in a
+ * document.
+ */
+export default class FeatureFinder {
+  constructor(private analyzer: Analyzer) {
```

Comment:
TBH I thought at first sight "this constructor doesn't do anything, it can be removed!" But then I realized this is a way to declare a private variable which will get set when a new class is instantiated...feels like magic!
No-op, just a note that this tricked me at first sight :p